mm2src/coins/eth/web3_transport.rs

```diff
@@ -193,30 +249,60 @@ async fn send_request(
 
         if !status.is_success() {
             errors.push(ERRL!(
-                "Server '{}' response !200: {}, {}",
-                uri,
+                "Server '{:?}' response !200: {}, {}",
+                node,
                 status,
                 binprint(&body, b'.')
             ));
             continue;
         }
 
-        return single_response(body, &uri.to_string());
+        return single_response(body, &node.uri.to_string());
     }
+
     Err(request_failed_error(&request, &errors))
 }
 
 #[cfg(target_arch = "wasm32")]
 async fn send_request(
     request: Call,
-    uris: Vec<http::Uri>,
+    nodes: Vec<Web3TransportNode>,
     event_handlers: Vec<RpcTransportEventHandlerShared>,
+    gui_auth_validation_generator: Option<GuiAuthValidationGenerator>,
 ) -> Result<Json, Error> {
-    let request_payload = to_string(&request);
+    let serialized_request = to_string(&request);
 
     let mut transport_errors = Vec::new();
-    for uri in uris {
-        match send_request_once(request_payload.clone(), &uri, &event_handlers).await {
+    for node in nodes {
+        let mut serialized_request = serialized_request.clone();
+        if node.gui_auth {
+            if let Some(generator) = gui_auth_validation_generator.clone() {
+                let signed_message = match EthCoin::generate_gui_auth_signed_validation(generator) {
+                    Ok(t) => t,
+                    Err(e) => {
+                        transport_errors.push(ERRL!(
+                            "GuiAuth signed message generation failed for {:?} node, error: {:?}",
+                            node,
+                            e
+                        ));
+                        continue;
+                    },
+                };
+
+                let auth_request = AuthPayload {
+                    request: &request,
+                    signed_message,
+                };
+
+                serialized_request = to_string(&auth_request);
+            } else {
+                transport_errors.push(ERRL!("GuiAuthValidationGenerator is not provided for {:?} node", node));
+                continue;
+            }
+        }
```

Comment:
How about moving this implementation to a separate function?

Reply:
> How about moving this implementation to a separate function?

Done